python/lsst/ts/tcpip/testone_client_read_loop_server.py

```diff
@@ -0,0 +1,59 @@
+# This file is part of ts_tcpip.
+#
+# Developed for the Rubin Observatory Telescope and Site System.
+# This product includes software developed by the LSST Project
+# (https://www.lsst.org).
+# See the COPYRIGHT file at the top-level directory of this distribution
+# for details of code ownership.
+#
+# This program is free software: you can redistribute it and/or modify
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation, either version 3 of the License, or
+# (at your option) any later version.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with this program.  If not, see <https://www.gnu.org/licenses/>.
+
+import asyncio
+import logging
+
+from .constants import LOCAL_HOST
+from .one_client_read_loop_server import OneClientReadLoopServer
+
+__all__ = ["TestOneClientReadLoopServer"]
+
+
+class TestOneClientReadLoopServer(OneClientReadLoopServer):
+    """An implementation of MessageReadingOneClientServer for unit test
+    purposes.
+    """
+
+    def __init__(self) -> None:
+        log = logging.getLogger()
+        super().__init__(
+            host=LOCAL_HOST,
+            port=0,
+            log=log,
+            name="MockMessageReadingOneClientServer",
+        )
+
+        self._message_queue: asyncio.Queue = asyncio.Queue()
+
+    async def read_and_dispatch(self) -> None:
+        message = await self.read_str()
+        await self._message_queue.put(message.strip())
```

Comment:
I gently suggest deleting the "strip" so that whatever read_str returns is what you get. This is especially important because `strip` strips leading and trailing whitespace as well as the terminator (at least for now -- if you change read_str to strip the terminator itself then it will not)

Reply:
Also please replace "message" with "data" (or vice-versa everywhere else), so the code is self-consistent).